core/src/main/java/org/apache/calcite/sql/SqlOverOperator.java

```diff
@@ -78,6 +78,12 @@ public SqlOverOperator() {
       throw validator.newValidationError(aggCall, RESOURCE.overNonAggregate());
     }
     final SqlNode window = call.operand(1);
+    SqlLiteral qualifier = aggCall.getFunctionQuantifier();
+    if (qualifier != null && qualifier.toString().equals("DISTINCT")
```

Comment:
I believe there should be some symbol defined DISTINCT, and if you reverse the equals operation you could save the `null` comparison

Reply:
I have changed it so that SqlSelectKeyword.DISTINCT is used instead of the literal "distinct". I still need to null check even with reversing of the equals operation since I would need to use qualifier.toString() or qualifier.getValue() and if qualifier is null it'll lead to NPE. Hence, I decided to keep it not-reversed for readability.